Source/core/Frame.h

```diff
@@ -303,59 +310,78 @@ namespace Core {
             }
 
         private:
-            uint16_t _offset;
+            SIZE_CONTEXT _offset;
             FrameType* _container;
         };
 
-    public:
+    public:       
         FrameType()
             : _size(0)
-            , _data()
-        {
+            , _data() {
+            // It looks like there is a bug in the windows compiler. It prepares a default/copy constructor
+            // if if the template being instantiated is not really utilizing it!
+            #ifndef __WINDOWS__
             static_assert(BLOCKSIZE != 0, "This method can only be called if you specify an initial blocksize");
+            #endif
         }
-        FrameType(const FrameType<BLOCKSIZE>& copy)
+        FrameType(const FrameType<BLOCKSIZE, BIG_ENDIAN_ORDERING, SIZE_CONTEXT>& copy) 
             : _size(copy._size)
             , _data(copy._data)
         {
+            // It looks like there is a bug in the windows compiler. It prepares a default/copy constructor
+            // if if the template being instantiated is not really utilizing it!
+            #ifndef __WINDOWS__
+            static_assert(BLOCKSIZE != 0, "This method can only be called if you allocate a new buffer");
+            #endif
         }
-        FrameType(uint8_t buffer[], const uint32_t length, const uint32_t loadedSize = 0)
+        FrameType(uint8_t* buffer, const SIZE_CONTEXT length, const SIZE_CONTEXT loadedSize = 0)
             : _size(loadedSize)
             , _data(buffer, length)
         {
             static_assert(BLOCKSIZE == 0, "This method can only be called if you pass a buffer that can not be extended");
         }
-        ~FrameType()
-        {
+        ~FrameType() = default;
+
+        FrameType<BLOCKSIZE, BIG_ENDIAN_ORDERING, SIZE_CONTEXT>& operator=(const FrameType<BLOCKSIZE, BIG_ENDIAN_ORDERING, SIZE_CONTEXT>& rhs) {
+            Size(rhs.Size());
```

Comment:
Add check for assigning to self? (behaviour memcpy in that case is undefined)

Reply:
added.